### ISSUE TITLE
Plugins for creating virtual tags

### ISSUE THREAD

**Issue by vpetzel** [quodlibet/quodlibet#3439]
Would it be possible to add a plugin interface for calculating additional tags?
E.g. due to ID3-conservative programs I have my music collection in such a way that multiple artists are in the interprets field (and on some operas with many performers the full cast in the comment). So I would create a plugin that splits the field and creates virtual performer tags. Or you might have some tracks with performer Kempff, some with Wilhelm Kempff and some with Kempff, Wilhelm or something, so you could do a plugin to add the performer Kempff to all of these. This would make using the browser for filtering for a specific artist much easier.

**Comment by declension** [comment]
Hard to tell from what you say, but sounds like a normal `SongsMenuPlugin`; you can then use `AudioFile`'s normal tag manipulation interface (well, actually `SongWrapper` really)

For a good example see [splitting.py](https://github.com/quodlibet/quodlibet/blob/master/quodlibet/ext/songsmenu/splitting.py)

**Comment by vpetzel** [comment]
I’m not sure if that works, for as I understand it a Songs Menu Plugin can only process tracks after selection. Wouldn’t you need something like an Event Plugin for that, to parse the songs as soon as they are added to the collection?

**Comment by declension** [comment]
Yes, ok, if you want to react to things happening, an Events Plugin is definitely what you want, though note very few (if any?) plugins do this, as there's not a good workflow management tooling here (see my old https://github.com/quodlibet/quodlibet/issues/275#issuecomment-269860578 actually) - e.g. what happens when you have two plugins doing things "on added". But for now this is probably a problem plugin users can live with
